src/TestableIO.System.IO.Abstractions.TestingHelpers/MockFile.cs

```diff
@@ -163,6 +163,11 @@ public override void Copy(string sourceFileName, string destFileName, bool overw
                 throw CommonExceptions.FileAlreadyExists(destFileName);
             }
 
+            if (string.Equals(sourceFileName, destFileName, StringComparison.OrdinalIgnoreCase) && XFS.IsWindowsPlatform())
```

Comment:
in all the other cases we use
`mockFileDataAccessor.StringOperations.Equals(sourceFileName, destinationFileName)` instead of relying on 
`string.Equals(..., ..., StringComparison.OrdinalIgnoreCase)`

Is intentional or an oversight?

Reply:
This seems to be an oversight.
We had one case [here](https://github.com/TestableIO/System.IO.Abstractions/pull/1256#discussion_r2050797816) where we used `string.Equals` on purpose, as it should compare case-sensitive in all operating systems, but here it seems I overlooked it.